packages/core/upload/server/services/upload.js

```diff
@@ -154,18 +188,39 @@ module.exports = ({ strapi }) => ({
 
   async upload({ data, files }, { user } = {}) {
```

Comment:
The idea behind my original PR is to allow users to upload buffers in the backend. Currently, the upload function wants a formdata object which is unfortunately only a [front end api](https://developer.mozilla.org/en-US/docs/Web/API/FormData). This implementation would still require the user to mimic that object structure by hand.

Reply:
The `upload` function doesn't specify that it's expecting formData. the `data` is a parsed body from the request and then files originally is the passed files from the context, but we can change that in the controller to be either `Buffers` from `body.file` or the `files`.